src/main/java/com/telefonica/iot/tidoop/backends/ckan/CKANBackend.java

```diff
@@ -0,0 +1,270 @@
+                    + "/api/3/action/organization_show?id=" + orgId;
+            CKANResponse resp = doCKANRequest("GET", url, "");
+            JSONObject result = (JSONObject) resp.getJsonObject().get("result");
+            JSONArray pkgs = (JSONArray) result.get("packages");
+            
+            for (Object pkg1 : pkgs) {
+                JSONObject pkg = (JSONObject) pkg1;
+                resIds.add((String) pkg.get("id"));
+            } // for
+            
+            return resIds;
+        } catch (Exception e) {
+            return null;
+        } // try catch
+    } // getPackages
+    
+    /**
+     * Gets the resource identifiers/names within a given package id/name.
+     * @param pkgId
+     * @return The resource identifiers/names within the given package id/name
+     */
+    public List<String> getResources(String pkgId) {
+        logger.debug("Getting the resources within " + pkgId + " + package");
+        
+        try {
+            List<String> resIds = new ArrayList<String>();
+            String url = "http" + (ssl ? "s" : "") + "://" + ckanHost + ":" + ckanPort
+                    + "/api/3/action/package_show?id=" + pkgId;
+            CKANResponse resp = doCKANRequest("GET", url, "");
+            JSONObject result = (JSONObject) resp.getJsonObject().get("result");
+            JSONArray resources = (JSONArray) result.get("resources");
+            
+            for (Object res : resources) {
+                JSONObject resource = (JSONObject) res;
+                resIds.add((String) resource.get("id"));
+            } // for
+            
+            return resIds;
+        } catch (Exception e) {
+            return null;
+        } // try catch
+    } // getResources
+    
+    /**
+     * Gets the total number of records within a given resource.
+     * @param resId Resource identifier
+     * @return The total number of records within the given resource
+     */
+    public int getNumRecords(String resId) {
+        logger.debug("Getting the number of records within " + resId);
+        
+        try {
+            int numRecords = 0;
+            int i = 0;
+            
+            while (true) {
+                String url = "http" + (ssl ? "s" : "") + "://" + ckanHost + ":" + ckanPort
+                        + "/api/3/action/datastore_search?limit=1000&offset=" + (i * 1000) + "&resource_id=" + resId;
+                CKANResponse resp = doCKANRequest("GET", url, "");
+                JSONObject result = (JSONObject) resp.getJsonObject().get("result");
+                JSONArray records = (JSONArray) result.get("records");
+                
+                if (records.size() == 0) {
+                    break;
+                } // if
+                
+                numRecords += records.size();
+                i++;
+            } // while
+            
+            return numRecords;
+        } catch (Exception e) {
+            return 0;
```

Comment:
Maybe some logging would be valuable to distinguish between the case when there are no records vs. the case when some exception occurred ;)